scripts/zkai_paper_claim_pack_gate.py

```
@@ -0,0 +1,397 @@
+            raise ClaimPackGateError(f"missing evidence path: {path}")
+        _assert_repo_contained(full_path, f"evidence_refs[{index}].path")
+
+
+def _assert_exact_list(value: Any, expected: list[str], label: str) -> None:
+    if value != expected:
+        raise ClaimPackGateError(f"{label} drift")
+
+
+def validate_payload(payload: dict[str, Any]) -> None:
+    if set(payload) != EXPECTED_KEYS:
+        raise ClaimPackGateError("unknown or missing claim-pack field")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)